src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -318,20 +319,15 @@ export const wrapWorkspace = ({ breadcrumbs, activeTab, title, topBarContent, sh
     const child = useRef()
 
     const { workspace, accessError, loadingWorkspace, storageDetails, refreshWorkspace } = useWorkspace(namespace, name)
-    const googleProject = workspace?.googleProject
-    const azureContext = workspace?.azureContext
-    const prevGoogleProject = usePrevious(googleProject)
-    const prevAzureContext = usePrevious(azureContext)
-
-    const workspaceLoaded = !!workspace
-
     const { runtimes, refreshRuntimes, persistentDisks, appDataDisks } = useCloudEnvironmentPolling(workspace)
-    const { apps, refreshApps } = useAppPolling(googleProject, name, workspace)
-    // The following if statements are necessary to support the context bar properly loading runtimes for google/azure
-    if (workspaceLoaded && (googleProject !== prevGoogleProject || azureContext !== prevAzureContext)) {
+    const { apps, refreshApps } = useAppPolling(workspace)
+
+    // The following is necessary to support the context bar properly loading runtimes for google/azure
+    useEffect(() => {
       refreshRuntimes(true)
-      refreshApps()
-    }
+      refreshApps(true)
+    }, [workspace]) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
It would be better to add `refreshRuntimes, refreshApps` to deps and remove the eslint disable-line.

Reply:
Weird, I do see "This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.". 

I would be concerned that this effect will be triggered much more often than you expect. Are you confident that it's only being called with the workspace itself changes? A react expert might be able to help out pretty quickly on this.

Reply:
From my testing it seems like the method is only be called when you switch tabs within the workspace, which I believe was the pre-existing behavior. So I will approve and you guys can sort this out later.

Reply:
Yeah.... it gets weird since `refreshApps` is technically a functional hook that returns a piece of its state. It is only called 3 times though, once when the workspace refreshes and then one other time (for reasons that aren't easily decipherable)

<img width="1391" alt="image" src="https://user-images.githubusercontent.com/6465084/217647131-e09da6f6-311d-4bfa-bda4-2133499e21d9.png">